cloudinfo/icdversions.go

```diff
@@ -0,0 +1,76 @@
+package cloudinfo
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/IBM/cloud-databases-go-sdk/clouddatabasesv5"
+	"github.com/IBM/go-sdk-core/v5/core"
+)
+
+type Deployable struct {
+	Type     string    `json:"type"`
+	Versions []Version `json:"versions"`
+}
+
+type Version struct {
+	Version string `json:"version"`
+	Status  string `json:"status"`
+}
+
+type Data struct {
+	Deployables []Deployable `json:"deployables"`
+}
+
+func (infoSvc *CloudInfoService) GetAvailableIcdVersions(icdType string) ([]string, error) {
+
+	authenticator := &core.IamAuthenticator{
+		ApiKey: infoSvc.authenticator.ApiKey, //pragma: allowlist secret
+	}
+	newOptions := &clouddatabasesv5.CloudDatabasesV5Options{
+		Authenticator: authenticator,
+	}
+
+	// Create the service client
+	service, err := clouddatabasesv5.NewCloudDatabasesV5(newOptions)
```

Comment:
This initialization of the clouddatabases service should be done in the NewCloudInfoSvc function, not here.